### DISCUSSION TITLE
How to use the web gui from comfyUI

### DISCUSSION THREAD

**Discussion by pardonmycode** [comfyanonymous/ComfyUI#5453] [Q&A]
I'm impressed by ComfyUI's web interface, particularly its node-based graph system, and would like to implement something similar in my project. However, I'm having difficulty understanding how to create such a node and graph system.
I've looked through the source code but haven't found clear examples of node creation and graph management, such as what I expected to find:
node = Node("Stable Diffusion", **kwargs)
graph.add(node)
Could someone explain how ComfyUI implements its node-graph system? Specifically, I'm interested in:

How nodes are created and structured ?
How the graph manages node connections ?
How the web interface visualizes these components ?

**Comment by LukeG89** [comment]
I'm not an expert, but I think it is [litegraph.js](https://github.com/Comfy-Org/litegraph.js)

Also check [ComfyUI_frontend](https://github.com/Comfy-Org/ComfyUI_frontend)

**Reply by pardonmycode** [reply]
Nice, thx.
